apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -535,6 +541,64 @@ func poolStorageProvider(poolManager poolmanager.PoolManager, registry storage.P
 	return providerType, pool.Attrs(), nil
 }
 
+func (a *API) applicationVolumeParams(
```

Comment:
I'm not sure this is right?
Does it conflict with applicationFilesystemParams?

Skipping the not provisioned volumes is making the assumption that the caller only wants to handle imported volumes. That might be ok, but we should ensure it's very clear that's what we're returning, eg the params.KubernetesVolumeParams could be called params.KubernetesImportedVolumeParams or something.

Reply:
TODO:

- A new API to query mongo base on the applicaition to find the detached storage.
- Only append storage which is detached.
- The name of the function should be rename to `getAttachableApplicationStorage`